src/Native/System.Native/pal_io.cpp

```diff
@@ -702,45 +703,57 @@ extern "C" int32_t FTruncate(int32_t fd, int64_t length)
     return ftruncate(fd, length);
 }
 
-void ConvertPollFDPalToPlatform(const PollFD& pal, pollfd& native)
+extern "C" Error Poll(PollEvent* pollEvents, uint32_t eventCount, int32_t milliseconds, uint32_t* triggered)
 {
-    native.fd = pal.FD;
-    native.events = pal.Events;
-    native.revents = pal.REvents;
-}
-
-static void ConvertPollFDPlatformToPal(const pollfd& native, PollFD& pal)
-{
-    pal.FD = native.fd;
-    pal.Events = native.events;
-    pal.REvents = native.revents;
-}
+    if (pollEvents == nullptr || triggered == nullptr)
+    {
+        return PAL_EFAULT;
+    }
 
-extern "C" int32_t Poll(PollFD* pollData, uint32_t numberOfPollFds, int32_t timeout)
-{
-    assert(numberOfPollFds <= 2);
+    if (milliseconds < -1)
+    {
+        return PAL_EINVAL;
+    }
 
-    if (numberOfPollFds > 2)
-        return ERANGE;
+    size_t bufferSize = sizeof(pollfd) * static_cast<size_t>(eventCount);
```

Comment:
Nit: logically it seems odd to cast eventCount to a size_t, since it's not a size.  Seems like it'd be better to cast the result of the multiplication to a size_t.